### PR TITLE
Add channel filter

### PR DESCRIPTION
**Pull request by oleksandr-didyk**

### CODE REVIEW

File: src/ProductConstructionService/ProductConstructionService.BarViz/Layout/NavMenu.razor [hunk 1]
```diff
@@ -12,12 +12,38 @@
...
+        </FooterContent>
+      </FluentAutocomplete>
+
       @foreach (var category in Categories)
       {
-        <FluentNavGroup Title="@category.Name" Icon="@(new Icons.Regular.Size20.List())">
+        <FluentNavGroup Title="@category.Name" Icon="@(new Icons.Regular.Size20.List())" Expanded="@expandedGroup">
           @foreach (var channel in category.Channels)
           {
-            <ChannelNavMenuItem Channel="@channel" />
```
Comment (oleksandr-didyk) [inline src/ProductConstructionService/ProductConstructionService.BarViz/Layout/NavMenu.razor:20]: My bad here, will revert; no idea how it got through

File: src/ProductConstructionService/ProductConstructionService.BarViz/Layout/NavMenu.razor [hunk 2]
```diff
@@ -12,12 +12,38 @@
 
       <h4 class="navmenu-heading">Channels</h4>
 
+      <FluentAutocomplete TOption="Channel"
+                          ImmediateDelay="10"
+                          AutoComplete="off"
+                          Autofocus="false"
+                          Placeholder="Select channel"
```
Comment (premun) [inline src/ProductConstructionService/ProductConstructionService.BarViz/Layout/NavMenu.razor:19]:
```suggestion
                          Placeholder="Filter channels"
```

File: src/ProductConstructionService/ProductConstructionService.BarViz/Layout/NavMenu.razor [hunk 3]
```diff
@@ -28,56 +54,91 @@
...
+        };
 
+        var otherCategory = new Category("Other");
         foreach (var channel in channels)
         {
             bool categorized = false;
-            foreach (var category in Categories)
+            foreach (var category in categories)
             {
                 if (channel.Name.StartsWith(category.Name))
```
Comment (premun) [inline src/ProductConstructionService/ProductConstructionService.BarViz/Layout/NavMenu.razor:100]: Can we add a category named "Test" and all categories with Classification equal "test" would go there? As a top priority rule?

Comment (oleksandr-didyk) [reply src/ProductConstructionService/ProductConstructionService.BarViz/Layout/NavMenu.razor:100]: Sure thing, on it

**Pull request merged**